src/runtime_src/core/tools/xbutil2/SubCmdValidate.cpp

```diff
@@ -1349,9 +1349,21 @@ get_platform_info(const std::shared_ptr<xrt_core::device>& device,
 {
   auto bdf = xrt_core::device_query<xrt_core::query::pcie_bdf>(device);
   ptTree.put("device_id", xrt_core::query::pcie_bdf::to_string(bdf));
-  ptTree.put("platform", xrt_core::device_query<xrt_core::query::rom_vbnv>(device));
-  ptTree.put("sc_version", xrt_core::device_query<xrt_core::query::xmc_sc_version>(device));
-  ptTree.put("platform_id", (boost::format("0x%x") % xrt_core::device_query<xrt_core::query::rom_time_since_epoch>(device)));
+
+  boost::property_tree::ptree platform_report;
+  boost::property_tree::ptree empty_ptree;
+  auto report = std::make_shared<ReportPlatforms>();
+  report->getPropertyTreeInternal(device.get(), platform_report);
+  
+  const boost::property_tree::ptree& platforms = platform_report.get_child("platforms", empty_ptree);
+  for(auto& kp : platforms) {
+    const boost::property_tree::ptree& pt_platform = kp.second;
+    const boost::property_tree::ptree& pt_static_region = pt_platform.get_child("static_region", empty_ptree);
+    ptTree.put("platform", pt_static_region.get<std::string>("vbnv"));
```

Comment:
If the child "static_region" doesn't exist, then the empty_ptree will be returned (which is good).

Unfortunately, "vbnv", "interface_uuid", and "controller.satellite_controller.version" will not be found and exceptions will be raised, which isn't so nice :(.

Wouldn't it be better if there was a` pt_static_region.empty() `check before retrieving this data?

Reply:
Yes, I missed those. Instead of checking for empty, the validation should continue. So I added `N/A` if any of the information is not found